lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -310,23 +310,32 @@ def _snap_daemons(
         except CalledProcessError as e:
             raise SnapError("Could not {} for snap [{}]: {}".format(args, self._name, e.stderr))
 
-    def get(self, key) -> str:
+    def get(self, key: str, use_json: bool = False) -> Any:
```

Comment:
I think instead of making it optional `key: Optional[str] = None`, it will be better to give a default value of empty string `key: str = ""`. Coincidentally, it will make it return all values for both `typed=True` or `typed=False`. Though, it might be useful for `typed=False` since it might looks something like this for nested return values:

```shell
$ sudo snap get openstack-hypervisor compute
Key                          Value
compute.cpu-mode             host-model
compute.spice-proxy-address  10.6.0.134
compute.virt-type            kvm
```